examples/lead-generation/src/graph/graph.ts

```diff
@@ -0,0 +1,41 @@
+import { CSV_GENERATOR_NODE_NAME, csvGeneratorNode } from "@/graph/nodes/csv-generator-node";
+import { ERROR_HANDLER, errorHandlerNode } from "@/graph/nodes/error-handler-node";
+import { OUTREACH_MESSAGE_NODE_NAME, outreachMessageNode } from "@/graph/nodes/outreach-message-node";
+import { ENRICH_THERAPISTS_NODE_NAME, enrichTherapistNode } from "@/graph/nodes/therapist-enrichment-node";
+import { FETCH_THERAPISTS, fetchTherapistsNode } from "@/graph/nodes/therapist-fetcher-node";
+import { URL_VALIDATOR, urlValidatorNode, validUrlCounterEdge } from "@/graph/nodes/url-validator-node";
+import { state } from "@/graph/state";
+import { END, START, StateGraph } from "@langchain/langgraph";
+
+export type LeadGenerationGraphResult = {
+  csvContent: string;
+  csvPath: string;
+  error: string;
+};
+
+export const leadGenerationGraph = async (graphInputs: string[]): Promise<LeadGenerationGraphResult> => {
+  const graphBuilder = new StateGraph(state)
+    .addNode(URL_VALIDATOR, urlValidatorNode)
+    .addNode(FETCH_THERAPISTS, fetchTherapistsNode)
+    .addNode(ENRICH_THERAPISTS_NODE_NAME, enrichTherapistNode)
+    .addNode(OUTREACH_MESSAGE_NODE_NAME, outreachMessageNode)
+    .addNode(CSV_GENERATOR_NODE_NAME, csvGeneratorNode)
+    .addNode(ERROR_HANDLER, errorHandlerNode);
+
+  // Edges
+  graphBuilder.addEdge(START, URL_VALIDATOR);
+  graphBuilder.addConditionalEdges(URL_VALIDATOR, validUrlCounterEdge, [FETCH_THERAPISTS, ERROR_HANDLER]);
+  graphBuilder.addEdge(FETCH_THERAPISTS, ENRICH_THERAPISTS_NODE_NAME);
+  graphBuilder.addEdge(ENRICH_THERAPISTS_NODE_NAME, OUTREACH_MESSAGE_NODE_NAME);
+  graphBuilder.addEdge(OUTREACH_MESSAGE_NODE_NAME, CSV_GENERATOR_NODE_NAME);
+  graphBuilder.addEdge(CSV_GENERATOR_NODE_NAME, END);
+  graphBuilder.addEdge(ERROR_HANDLER, END);
+
+  const graph = graphBuilder.compile();
+
+  const result = await graph.invoke({
```

Comment:
Related to Theo's comments around the Airtop client we can use a the `config` argument when invoking the graph to pass in runtime variables (like the AirtopClient instance) and it will get passed to every node (in the 2nd argument to the node) when it executes. We could use this to isolate the api key/AirTop client instance initialization when we're defining the graph here and avoid having to do the client initialization in each node. Check out this example from their docs for more on how they're using the config argument: https://langchain-ai.github.io/langgraphjs/how-tos/pass-run-time-values-to-tools/?h=context#use-it